app/preview.py

```diff
@@ -98,31 +98,30 @@ def view_letter_template(filetype):
     pdf = get_pdf(html)
 
     letter_attachment = json["template"].get("letter_attachment", {})
-    if filetype == "pdf":
-        if letter_attachment:
-            pdf = add_attachment_to_letter(
-                service_id=json["template"]["service"], templated_letter_pdf=pdf, attachment_object=letter_attachment
-            )
+    if letter_attachment:
+        pdf = add_attachment_to_letter(
+            service_id=json["template"]["service"], templated_letter_pdf=pdf, attachment_object=letter_attachment
+        )
 
+    if filetype == "pdf":
         return send_file(
             path_or_file=pdf,
             mimetype="application/pdf",
         )
+
     elif filetype == "png":
-        templated_letter_page_count = get_page_count(pdf)
+        # get pdf that can be read multiple times - unlike StreamingBody from boto that can only be read once
+        pdf_persist = BytesIO(pdf) if isinstance(pdf, bytes) else BytesIO(pdf.read())
```

Comment:
We are sometimes sending the StreamingBody as is - if the request hitting the endpoint asks for a pdf, and it's a basic letter (no attachment, not bilingual), and it was in the cache. But since both StreamingBody and the other option are received well, I think we can try what you suggest - it will be nice to have consistent format to work with in this view. I will try this today and let you know how I get on 😁

Reply:
Ah, if we are using the streaming response then in some places then it would be nice to keep that 🤔  Feel free to leave as-is 🤔